esprima.js

```diff
@@ -674,6 +677,19 @@
 
         // Check for most common single-character punctuators.
         case 0x2E:  // . dot
+            ch3 = source.substr(index, 3);
```

Comment:
In theory, `charCodeAt` can be much faster but probably vms are not smart enough to optimize it. As I tested, there is no significant differences in numbers.